clientapi/routing/room_tagging.go

```diff
@@ -0,0 +1,208 @@
+// Copyright 2019 Sumukha PK
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package routing
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/matrix-org/dendrite/clientapi/auth/storage/accounts"
+	"github.com/matrix-org/dendrite/clientapi/httputil"
+	"github.com/matrix-org/gomatrix"
+	"github.com/matrix-org/gomatrixserverlib"
+	"github.com/matrix-org/util"
+)
+
+// newTag creates and returns a new Tag type
+func newTag() gomatrix.TagContent {
+	return gomatrix.TagContent{
+		Tags: make(map[string]gomatrix.TagProperties),
+	}
+}
+
+// GetTag implements GET /_matrix/client/r0/user/{userID}/rooms/{roomID}/tags
+func GetTag(
+	req *http.Request,
+	accountDB *accounts.Database,
+	userID string,
+	roomID string,
+) util.JSONResponse {
```

Comment:
Careful! We [need](https://matrix.org/docs/spec/client_server/r0.5.0#get-matrix-client-r0-user-userid-rooms-roomid-tags) to make sure:

> The access token must be authorized to make requests for this user ID.

So a `device.UserID == userID` check here would be necessary. See [sendtyping.go](https://github.com/matrix-org/dendrite/blob/master/clientapi/routing/sendtyping.go#L40) for an example.

This is the same for other similar functions.

Reply:
Thanks for pointing this out!
Agreed, I had missed it.